src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -167,10 +167,11 @@ private IReadOnlyCollection<IHttpCookie> ToHttpCookies(string cookieString)
 
             for (int c = 0; c < separateCookies.Length; c++)
             {
-
-                var splitArray = separateCookies[c].Split(separator, StringSplitOptions.RemoveEmptyEntries);
+                // count:2 tells the split method to stop splitting after
```

Comment:
From the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=net-5.0#system-string-split(system-string-system-int32-system-stringsplitoptions)),

> If the string has already been split count - 1 times, but the end of the string has not been reached, then the last string in the returned array will contain this instance's remaining trailing substring, untouched.

Using of this overload will handle string like "token=a12=" where token is the cookie name and a12= is the cookie value.